pyswarms/utils/decorators/decorators.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+
+def cost(cost_func):
+    """A decorator to mark the cost function
+
+    This decorator provides a way to simplify the cost function. The cost function
+    needs to resemble a strategy that can be applied to every particle in the swarm
+    to assign a cost. This means that the decorated cost function must use an array
+    of shape :code:`(dimensions, )` as an argument and return a :code:`float` or an
+    :code:`int`.
```

Comment:
I think this can be phrased as:

> If you have defined a cost function that only acts upon a single particle, then you can use this decorator to have a vectorized computation for *n* particles. ...

Then give an example `code-block` maybe? I think good keywords here are: *wrapper*, *transforms*, etc. *Strategy* seems a bit odd (I think you got this from hypothesis haha but I think it's more of their local lingo).